history.php

```diff
@@ -21,17 +21,36 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
+use tool_clonecategory\form\clonecategoryhistory_form;
 use tool_clonecategory\history_table;
 
 require('../../../config.php');
 require_once($CFG->libdir . '/adminlib.php');
 
 admin_externalpage_setup('clonecategory_history');
 
+$start = optional_param('start', strtotime('-3 month', time()), PARAM_INT);
+$end = optional_param('end', time(), PARAM_INT);
+
+$clonehistoryform = new clonecategoryhistory_form(null, [
+   'startdate' => $start,
+   'enddate' => $end,
+]);
+
+// Redirect back if form is cancelled.
+if ($clonehistoryform->is_cancelled()) {
+    redirect(new moodle_url("/admin/tool/clonecategory/history.php"));
+}
+
 $PAGE->set_url(new moodle_url("/admin/tool/clonecategory/history.php"));
 $PAGE->set_context(context_system::instance());
 
 echo $OUTPUT->header();
 echo $OUTPUT->heading(get_string('history_table', 'tool_clonecategory'));
-history_table::display();
+
+if ($data = $clonehistoryform->get_data()) {
+    history_table::display($data->startdate, $data->enddate);
+}
+$clonehistoryform->display();
```

Comment:
I feel like the form should be displayed above the table. So that if the table is really long, you don't have to scroll to the bottom to reach it.

Reply:
moved above in latest commit